nestkernel/source_table.cpp

```diff
@@ -408,3 +448,103 @@ nest::SourceTable::get_next_target_data( const thread tid,
     return true; // found a valid entry
   }
 }
+
+void
+nest::SourceTable::resize_compressible_sources()
+{
+  for ( thread tid = 0; tid < static_cast< thread >( compressible_sources_.size() ); ++tid )
+  {
+    compressible_sources_[ tid ].clear();
+    compressible_sources_[ tid ].resize(
+      kernel().model_manager.get_num_synapse_prototypes(), std::map< index, SpikeData >() );
+  }
+}
+
+void
+nest::SourceTable::collect_compressible_sources( const thread tid )
+{
+  for ( synindex syn_id = 0; syn_id < sources_[ tid ].size(); ++syn_id )
+  {
+    index lcid = 0;
+    while ( lcid < sources_[ tid ][ syn_id ].size() )
+    {
+      const std::pair< index, SpikeData > source_node_id_to_spike_data =
+        std::make_pair( sources_[ tid ][ syn_id ][ lcid ].get_node_id(), SpikeData( tid, syn_id, lcid, 0 ) );
+      compressible_sources_[ tid ][ syn_id ].insert( source_node_id_to_spike_data );
+
+      // find next source with different node_id (assumes sorted sources)
+      const index old_source_node_id = sources_[ tid ][ syn_id ][ lcid ].get_node_id();
```

Comment:
If you move this before the `make_pair` line, you can save one call to `get_node_id()` and improve clarity.

Reply:
ok